src/plugins/intel_cpu/src/cpu_shape.h

```diff
@@ -19,24 +19,25 @@ class Shape {
     Shape() = default;
 
     explicit Shape(const ov::PartialShape& shape) {
-        minDims = shape.get_min_shape();
-        std::transform(minDims.begin(), minDims.end(), minDims.begin(), [](Dim x){ return ov::Interval::s_max == x ? UNDEFINED_DIM : x;});
-        maxDims = shape.get_max_shape();
-        std::transform(maxDims.begin(), maxDims.end(), maxDims.begin(), [](Dim x){ return ov::Interval::s_max == x ? UNDEFINED_DIM : x;});
-        type = shape.is_static() ? ShapeType::Static : ShapeType::Dynamic;
+        if (!shape.rank().is_dynamic()) {
+            minDims.reserve(shape.rank().get_length());
+            maxDims.reserve(shape.rank().get_length());
```

Comment:
Detail: Consider create `shape_rank` and use it as `rank()` creates new Rank object.
```suggestion
        const auto shape_rank = shape.rank();
        if (!shape_rank.is_dynamic()) {
            minDims.reserve(shape_rank.get_length());
            maxDims.reserve(shape_rank.get_length());
```